src/rust/ide/view/graph-editor/src/component/visualization/foreign/java_script/binding.rs

```diff
@@ -86,10 +94,57 @@ impl JsConsArgs {
         self.root.clone().into()
     }
 
+    /// Getter for the theming API that we expose to JS visualizations
+    pub fn theme(&self) -> JsTheme {
+        self.theme.clone()
+    }
+
     /// Helper method to emit an preprocessor change event from the visualisation.
     pub fn emit_preprocessor_change(&self, code:Option<String>, module:Option<String>){
         let closure             = &self.set_preprocessor;
         let preprocessor_config = PreprocessorConfiguration::from_options(code,module);
         (*closure)(preprocessor_config);
     }
 }
+
+/// The theming API that we expose to JS visualizations
+#[wasm_bindgen]
+#[derive(Clone, Debug)]
+pub struct JsTheme {
+    styles: StyleWatch
+}
+
+#[allow(non_snake_case)]
+#[wasm_bindgen]
+impl JsTheme {
+    /// Takes a qualified type name and returns the color that is used in the GUI for that type.
+    pub fn getColorForType(&self, tp_name: &str) -> String {
+        let tp = Type::from(tp_name.to_string());
+        let lcha = type_coloring::compute(&tp,&self.styles);
+        format_lcha(lcha)
+    }
+
+    /// Takes a qualified type name and returns the color that should be used for foreground
+    /// (e.g. text) that is shown on top of the background color returned by getColorForType.
+    pub fn getForegroundColorForType(&self, _tp_name: &str) -> String {
+        "white".to_string()
+    }
+
+    /// Queries style sheet value for a value.
+    pub fn get(&self, path: &str) -> Option<String> {
+        if let style::Data::Color(lcha) = self.styles.get(path)? {
+            Some(format_lcha(lcha))
+        } else {
+            None
+        }
+    }
+}
+
+fn format_lcha(lcha: color::Lcha) -> String {
```

Comment:
the name `format_lcha` is not semantic - it doesn't tell what it does. if you want to refactor it to a separate function it should be named `lcha_to_javascript_string`. However, after some changes proposed in this PR this would not be needed anymore, as you would be able to do `color.to_rgba().to_javascript_string()` instead.